bliss/metrics.py

```diff
@@ -40,50 +43,60 @@ def __init__(
             fp: false positives = # of predicted sources not matched with true source
             avg_distance: Average l-infinity distance over matched objects.
             total_true_n_sources: Total number of true sources over batches seen.
-            total_correct_class: Total # of correct classifications over matched objects.
-            total_n_matches: Total # of matches over batches.
-            conf_matrix: Confusion matrix (galaxy vs star)
             disable_bar: Whether to show progress bar
+            gal_tp: true positives = # of sources correctly classified as a galaxy
+            gal_fp: false positives = # of sources incorrectly classified as a galaxy
+            gal_fn: false negatives = # of sources incorrectly classified as a star
+            gal_tn: true negatives = # of sources correctly classified as a star
         """
         super().__init__(dist_sync_on_step=dist_sync_on_step)
 
         self.slack = slack
         self.disable_bar = disable_bar
 
         self.add_state("tp", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("tp_gal", default=torch.tensor(0), dist_reduce_fx="sum")
         self.add_state("fp", default=torch.tensor(0), dist_reduce_fx="sum")
         self.add_state("avg_distance", default=torch.tensor(0.0), dist_reduce_fx="sum")
         self.add_state("total_true_n_sources", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("total_correct_class", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("conf_matrix", default=torch.tensor([[0, 0], [0, 0]]), dist_reduce_fx="sum")
+        self.add_state("gal_tp", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_fp", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_fn", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_tn", default=torch.tensor(0), dist_reduce_fx="sum")
 
     # pylint: disable=no-member
     def update(self, true: FullCatalog, est: FullCatalog) -> None:  # type: ignore
-        """Update the internal state of the metric including tp, fp, total_true_n_sources, etc."""
+        """Update the internal state of metrics including tp, fp, total_coadd_gal_matches, etc."""
         assert true.batch_size == est.batch_size
 
         count = 0
-        desc = "Detection Metric per batch"
+        desc = "Bliss Metric per batch"
```

Comment:
Can/should we get rid of tqdm on the line before this one? Presumably the loop will be extremely fast -- almost instant

Reply:
Sure - I wasn't sure if the progress bar was needed for some reason. Sounds good!